pkg/diagnostics/diagnostics.go

```diff
@@ -49,10 +56,23 @@ func (r *Registry) RegisterConnectedPeersSource(
 	r.Registry.RegisterSource("connected_peers", func() string {
 		connectionManager := netProvider.ConnectionManager()
 		connectedPeers := connectionManager.ConnectedPeers()
+		connectedPeersAddrInfo := connectionManager.ConnectedPeersAddrInfo()
 
 		peersList := make([]map[string]interface{}, len(connectedPeers))
 		for i := 0; i < len(connectedPeers); i++ {
 			peer := connectedPeers[i]
+			peersAddrInfo := connectedPeersAddrInfo[i].Addrs
+
+			var peerAddr string
+			for _, peerAddrInfo := range peersAddrInfo {
+				if strings.Contains(peerAddrInfo.String(), eligibleDNS4) ||
+					(strings.Contains(peerAddrInfo.String(), eligibleIP4)) {
```

Comment:
Why do we filter out everything that is not ip4 or dns4? I think it's possible that we can get someone connected via ip6. 

I think we should return all addresses known by the peer store, and if we want to do some filtering, we should do it in the code consuming the output of this endpoint.

Reply:
All the protocol codes are here: https://github.com/multiformats/multiaddr#protocols I don't think we should return all of them. After talking to @nkuba we decided to focus on ip4 and dns4. If you think we should also allow for ip6 then I can add it as well.

Reply:
How are these protocol codes assembled? Aren't they resolved by the peerstore based on how the client can be reached?

Reply:
I've timeboxed my search, and this is the closest what I found to answer your question
https://github.com/libp2p/specs/blob/master/addressing/README.md#transport-multiaddrs

My understanding is that libp2p supports some of the `multiaddr` protocols described in this table https://github.com/multiformats/multiaddr/blob/master/protocols.csv The docs says 
> Most libp2p transports use the IP protocol as a foundational layer, and as a result, most transport multiaddrs will begin with a component that represents an IPv4 or IPv6 address.
Also reading further, it seems that libp2p will try to resolve protocols to IPv4 or IPv6. I'm not sure if there are any edge cases when libp2p can't/won't resolve to ip4/ip6 but this lib is not trivial and it's hard to answer. 

Having said that, in most cases I think we should be fine without any filtering just like you suggested, but I'm not 100% confident.

Reply:
Here we have confirmation in the code:

> libp2p makes a distinction between a peer's identity and its location. A peer's identity is stable, verifiable, and valid for the entire lifetime of the peer (whatever that may be for a given application). Peer identities are derived from public keys as described in the [peer id spec](https://github.com/libp2p/specs/blob/master/peer-ids/peer-ids.md).
> 
> On a particular network, at a specific point in time, a peer may have one or more locations, which can be represented using addresses. For example, I may be reachable via the global IPv4 address of 7.7.7.7 on TCP port 1234.
> 
> In a system that only supported TCP/IP or UDP over IP, we could easily write our addresses with the familiar <ip>:<port> notation and store them as tuples of address and port. However, libp2p was designed to be transport agnostic, which means that we can't assume that we'll even be using an IP-backed network at all.

(from https://github.com/libp2p/specs/blob/master/addressing/README.md#overview)

Now, our `libp2p.go`:

https://github.com/keep-network/keep-core/blob/723bc65b9a46914a333d6f8a317ada4cf8b8d099/pkg/net/libp2p/libp2p.go#L400-L405

From [libp2p `options.go`](https://github.com/libp2p/go-libp2p/blob/b54bacef92dd04d034cc2608364499d015c80f60/options.go#L44-L50):

```
// ListenAddrs configures libp2p to listen on the given addresses.
func ListenAddrs(addrs ...ma.Multiaddr) Option {
	return func(cfg *Config) error {
		cfg.ListenAddrs = append(cfg.ListenAddrs, addrs...)
		return nil
	}
}
```

And then, in [`libp2p.go` documentation for `func New(opts ...Option) (host.Host, error)`](https://github.com/libp2p/go-libp2p/blob/b54bacef92dd04d034cc2608364499d015c80f60/libp2p.go#L34-L38):
```
// - If no transport and listen addresses are provided, the node listens to
// the multiaddresses "/ip4/0.0.0.0/tcp/0" and "/ip6/::/tcp/0";
//
// - If no transport options are provided, the node uses TCP, websocket and QUIC
// transport protocols;
```

Reply:
Makes sense. Also, worth noting that this part 
```
// - If no transport options are provided, the node uses TCP, websocket and QUIC
// transport protocols;
```
leads to [TCP](https://github.com/libp2p/specs/blob/master/addressing/README.md#tcp), [WebSockets](https://github.com/libp2p/specs/blob/master/addressing/README.md#websockets) and [QUIC](https://github.com/libp2p/specs/blob/master/addressing/README.md#quic) which all of them resolve to `<ip-multiaddr>` which is described in [IP and Name resolution](https://github.com/libp2p/specs/blob/master/addressing/README.md#ip-and-name-resolution)